portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -30,44 +30,45 @@
-        long timestamp = System.currentTimeMillis();
-        double sentimentScore = getSentimentScore(message);
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String name = request.getParameter("name-input");
+    String email = request.getParameter("email-input");
+    String subject = request.getParameter("subject-input");
+    String message = request.getParameter("message-input");
+    long timestamp = System.currentTimeMillis();
 
-        // Error handling - don't allow empty or null values
-        if (!Strings.isNullOrEmpty(name) && !Strings.isNullOrEmpty(email) && !Strings.isNullOrEmpty(subject) && !Strings.isNullOrEmpty(message)) {
-            Entity commentEntity = new Entity("Comment");
-            commentEntity.setProperty("name", name);
-            commentEntity.setProperty("email", email);
-            commentEntity.setProperty("subject", subject);
-            commentEntity.setProperty("message", message);
-            commentEntity.setProperty("timestamp", timestamp);
-            commentEntity.setProperty("sentimentScore", sentimentScore);
+    LanguageServiceClient languageService = LanguageServiceClient.create();
+    double sentimentScore = getSentimentScore(message, languageService);
+    languageService.close();
 
-            DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-            datastore.put(commentEntity);
-        }
+    // Error handling - don't allow empty or null values
+    if (!Strings.isNullOrEmpty(name) && !Strings.isNullOrEmpty(email) && !Strings.isNullOrEmpty(subject) && !Strings.isNullOrEmpty(message)) {
+      Entity commentEntity = new Entity("Comment");
+      commentEntity.setProperty("name", name);
+      commentEntity.setProperty("email", email);
+      commentEntity.setProperty("subject", subject);
+      commentEntity.setProperty("message", message);
+      commentEntity.setProperty("timestamp", timestamp);
+      commentEntity.setProperty("sentimentScore", sentimentScore);
 
-        // Redirect back to the HTML page.
-        response.sendRedirect("/index.html");
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(commentEntity);
     }
 
-    double getSentimentScore(String message) throws IOException {
-        Document doc = Document.newBuilder().setContent(message).setType(Document.Type.PLAIN_TEXT).build();
-		LanguageServiceClient languageService = LanguageServiceClient.create();
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
+  double getSentimentScore(String message, LanguageServiceClient languageService) throws IOException {
+    Document doc = Document.newBuilder().setContent(message).setType(Document.Type.PLAIN_TEXT).build();
 		Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
```

Comment:
Something still seems off with the indentation...are tabs being used instead of spaces?

Reply:
Im not sure why this keeps changing back when i push it?! I swear im changing it locally lol